step-templates/deploy-child-project.json

```diff
@@ -1,15 +1,15 @@
TenantedDeploymentMode -ne \"Untenanted\")\n{\n    $createdRelease.TenantId = $deploymentTenantId\n}\n\nif ($whatIf -eq $true)\n{    \t\n    Write-OctopusVerbose \"Would have done a POST to /api/$spaceId/deployments with the body:\"\n    Write-OctopusVerbose $($createdRelease | ConvertTo-JSON)        \n    \n    Write-OctopusSuccess \"What If set to true.\"\n    Write-OctopusSuccess \"Setting the output variable ReleaseToPromote to $($releaseToPromote.Release.Version).\"            \n\tSet-OctopusVariable -Name \"ReleaseToPromote\" -Value ($releaseToPromote.Release.Version)                   \n}\n\nWrite-OctopusVerbose \"Getting the release notes\"\n$releaseNotes = Get-ReleaseNotes -releaseToPromote $releaseToPromote -deploymentPreview $deploymentPreview -channel $channel -spaceId $spaceId -defaultUrl $defaultUrl -octopusApiKey $octopusApiKey\nWrite-OctopusSuccess \"Setting the output variable ReleaseNotes which contains the release notes from the child project\"\nSet-OctopusVariable -Name \"ReleaseNotes\" -value $releaseNotes\n\nTest-ChildProjectDeploymentCanProceed -releaseToPromote $releaseToPromote -allowRedeployToTargetEnvironment $allowRedeployToTargetEnvironment -specificMachineDeployment $specificMachineDeployment -releaseNumber $releaseNumber -environmentName $environmentName -childDeploymentSpecificMachines $childDeploymentSpecificMachines -project $project -ignoreSpecificMachineMismatch $ignoreSpecificMachineMismatch -deploymentMachines $deploymentMachines\n\nif ($saveReleaseNotesAsArtifact -eq $true)\n{\n\t$releaseNotes | Out-File \"ReleaseNotes.txt\"\n    $currentDate = Get-Date\n\t$currentDateFormatted = $currentDate.ToString(\"yyyy_MM_dd_HH_mm\")\n    $artifactName = \"$($project.Name) $($releaseToPromote.Release.Version) From $($previousPhase.Name) To $($targetPhase.Name).ReleaseNotes_$($currentDateFormatted).txt\"\n    Write-OctopusInformation \"Creating the artifact $artifactName\"\n    \n\tNew-OctopusArtifact -Path \"ReleaseNotes.txt\" -Name $artifactName\n}\n\nif ($whatif -eq $true)\n{\n\tWrite-OctopusSuccess \"Exiting because What If set to true.\"\n    Exit 0\n}\n\n$parentDeploymentApprovers = Get-ParentDeploymentApprovers -parentDeploymentTaskId $parentDeploymentTaskId -spaceId $spaceId -defaultUrl $DefaultUrl -octopusApiKey $octopusApiKey\n\nCreate-NewOctopusDeployment -releaseToPromote $releaseToPromote -targetEnvironment $targetEnvironment -createdRelease $createdRelease -project $project -waitForFinish $waitForFinish -deploymentCancelInSeconds $deploymentCancelInSeconds -defaultUrl $defaultUrl -octopusApiKey $octopusApiKey -spaceId $spaceId -parentDeploymentApprovers $parentDeploymentApprovers -parentProjectName $parentProjectName -parentReleaseNumber $parentReleaseNumber -parentEnvironmentName $parentEnvironmentName -parentDeploymentTaskId $parentDeploymentTaskId -parentProjectSlug $parentProjectSlug -parentDeploymentId $parentDeploymentId -autoapproveChildManualInterventions $autoapproveChildManualInterventions\n\n\n"
```

Comment:
A couple of minor issues:
- `Write-OctopusInformation "SpaceId: $spaceId"` - I think `$spaceId` should be `$destinationSpaceId` as these log lines are written before `$spaceId` is assigned.
- Two ocurrences of `$relesaeNotes += ""` when building the release notes